### PR TITLE
Tweak Accessor Generator

### PR DESCRIPTION
**Pull request by agroGandhi**
This PR adds a mechanism to generate the JustTweak integration code.
The original idea goes back to #47 when we realised we could improve the library to avoid, by design, potential inconsistencies.

The whole work has been carried out by @agroGandhi and @albertodebortoli in a [public fork](https://github.com/agroGandhi/JustTweak) producing a number of pull requests:
- https://github.com/agroGandhi/JustTweak/pull/1
- https://github.com/agroGandhi/JustTweak/pull/2
- https://github.com/agroGandhi/JustTweak/pull/3
- https://github.com/agroGandhi/JustTweak/pull/4
- https://github.com/agroGandhi/JustTweak/pull/5
- https://github.com/agroGandhi/JustTweak/pull/6
- https://github.com/agroGandhi/JustTweak/pull/7
- https://github.com/agroGandhi/JustTweak/pull/8

This PR includes all the above pull requests and for this reason is quite large. We recommend the reviewers to give the project a try to fully grasp what the changes are about.

In a nutshell, JustTweak has now the ability to create the integration code for you at compile time based on a simple configuration file, hugely simplifying the usage of the component.

We approached this with a fresh mind, taking the opportunity to rename symbols (e.g. `Configuration` is now named `TweakProvider`) causing the need for a major version bump (version 6) once published.
Please refer to the README for more info.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -79,31 +184,31 @@ if enabled {
...
 ```
 
 `@TweakProperty` and `@OptionalTweakProperty` property wrappers are available to mark properties representing feature flags. Mind that by using these property wrappers, a static instance of `TweakManager` is needed. 
 
-```
+```swift
 @TweakProperty(fallbackValue: <#fallback_value#>,
                feature: <#feature_key#>,
                variable: <#variable_key#>,
                tweakManager: <#TweakManager#>)
```
Comment (aderussell) [inline README.md:207]: This needs to be updated to `@FallbackTweakProperty` or have the `fallbackValue` param removed from the signature

File: TweakAccessorGenerator/TweakAccessorGenerator/Extensions/NSNumber+ValueTypes.swift [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        switch encoding {
+        case "d":
+            return "Double"
+                
+        case "q":
+            return "Int"
+            
+        case "c":
+            return "Bool"
+
```
Comment (aderussell) [inline TweakAccessorGenerator/TweakAccessorGenerator/Extensions/NSNumber+ValueTypes.swift:21]: This method is not comprehensive as `q` will apply for a signed long long, not all forms of int within an NSNumber. `c` could also be any signed char, not just a boolean. 
Are you happy with this?

**Review by aderussell — CHANGES_REQUESTED** [review]

**Review by gigisommo — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (albertodebortoli) [reply README.md:207]: Well spotted! Thanks.

File: TweakAccessorGenerator/TweakAccessorGenerator/Extensions/NSNumber+ValueTypes.swift (hunk 2 above)
Comment (albertodebortoli) [reply TweakAccessorGenerator/TweakAccessorGenerator/Extensions/NSNumber+ValueTypes.swift:21]: Yes, I'm happy with this, it's annoying in the first place we have to do this. We had this already in JustTweak, in fact it was copied from [here](https://github.com/justeat/JustTweak/blob/master/JustTweak/Classes/Extensions/TweakExtensions.swift#L23). Those cases and implementation should be enough, I think we have enough tests to give us confidence.

**Review by aderussell — APPROVED** [review]
